crates/wasmparser/src/validator/component.rs

```diff
@@ -2503,6 +2511,14 @@ impl ComponentState {
             }
         }
 
+        let string_encoding = match encoding {
```

Comment:
Instead of `encoding: Option<CanonicalOption>`, could `encoding: Option<StringEncoding>` be the local variable?